LibrariesTest fail to verify https://dialogflow.com

Currently all tests are suddenly failing with the following error:

```
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 51.686 s <<< FAILURE! - in com.google.cloud.tools.libraries.LibrariesTest
[ERROR] testJson(com.google.cloud.tools.libraries.LibrariesTest)  Time elapsed: 51.638 s  <<< ERROR!
java.net.ProtocolException: Server redirected too many  times (20)
        at com.google.cloud.tools.libraries.LibrariesTest.assertReachable(LibrariesTest.java:128)
        at com.google.cloud.tools.libraries.LibrariesTest.assertApi(LibrariesTest.java:88)
        at com.google.cloud.tools.libraries.LibrariesTest.testJson(LibrariesTest.java:69)
```

Turns out `HttpURLConnection` always fails to connect to  `dialogflow.com`.

```java
// This always fails.
((HttpURLConnection) new URL("https://dialogflow.com").openConnection()).getResponseCode();
```

I figured out it does multiple complex redirections bouncing back and forth between dialogflow.com and accounts.google.com. And the redirection never ends successfully without setting cookies required by the servers. Fortunately, there's a simple way to enable automatic cookie managing.

--- a/src/test/java/com/google/cloud/tools/libraries/LibrariesTest.java
+++ b/src/test/java/com/google/cloud/tools/libraries/LibrariesTest.java
@@ -50,6 +50,7 @@ public class LibrariesTest {
   @Before
   public void parseJson() {
     oldCookieHandler = CookieHandler.getDefault();
+    // https://github.com/GoogleCloudPlatform/appengine-plugins-core/issues/822
     CookieHandler.setDefault(new CookieManager());
 
     JsonReaderFactory factory = Json.createReaderFactory(null);

--- a/src/test/java/com/google/cloud/tools/libraries/LibrariesTest.java
+++ b/src/test/java/com/google/cloud/tools/libraries/LibrariesTest.java
@@ -22,6 +22,8 @@ import com.google.common.collect.Maps;
 import com.google.common.collect.Sets;
 import java.io.IOException;
 import java.io.InputStream;
+import java.net.CookieHandler;
+import java.net.CookieManager;
 import java.net.HttpURLConnection;
 import java.net.URL;
 import java.util.ArrayList;
@@ -35,6 +37,7 @@ import javax.json.JsonObject;
 import javax.json.JsonReader;
 import javax.json.JsonReaderFactory;
 import javax.json.JsonString;
+import org.junit.After;
 import org.junit.Assert;
 import org.junit.Before;
 import org.junit.Test;
@@ -42,13 +45,22 @@ import org.junit.Test;
 public class LibrariesTest {
 
   private JsonObject[] apis;
+  private CookieHandler oldCookieHandler;
 
   @Before
   public void parseJson() {
+    oldCookieHandler = CookieHandler.getDefault();
+    CookieHandler.setDefault(new CookieManager());
+
     JsonReaderFactory factory = Json.createReaderFactory(null);
     InputStream in = LibrariesTest.class.getResourceAsStream("libraries.json");
     JsonReader reader = factory.createReader(in);
     apis = reader.readArray().toArray(new JsonObject[0]);
+  }
+
+  @After
+  public void tearDown() {
+    CookieHandler.setDefault(oldCookieHandler);
   }
 
   @Test

--- a/src/test/java/com/google/cloud/tools/managedcloudsdk/ManagedCloudSdkTest.java
+++ b/src/test/java/com/google/cloud/tools/managedcloudsdk/ManagedCloudSdkTest.java
@@ -44,7 +44,7 @@ public class ManagedCloudSdkTest {
 
   @Rule public TemporaryFolder tempDir = new TemporaryFolder();
 
-  private static final String FIXED_VERSION = "200.0.0";
+  private static final String FIXED_VERSION = "272.0.0";
   private static final Map<String, String> EMPTY_MAP = Collections.emptyMap();
   private final MessageCollector testListener = new MessageCollector();
   private final ProgressListener testProgressListener = new NullProgressListener();

--- a/src/test/java/com/google/cloud/tools/managedcloudsdk/ManagedCloudSdkTest.java
+++ b/src/test/java/com/google/cloud/tools/managedcloudsdk/ManagedCloudSdkTest.java
@@ -44,7 +44,7 @@ public class ManagedCloudSdkTest {
 
   @Rule public TemporaryFolder tempDir = new TemporaryFolder();
 
-  private static final String FIXED_VERSION = "272.0.0";
+  private static final String FIXED_VERSION = "276.0.0";
   private static final Map<String, String> EMPTY_MAP = Collections.emptyMap();
   private final MessageCollector testListener = new MessageCollector();
   private final ProgressListener testProgressListener = new NullProgressListener();

